pkg/target/target.go

```diff
@@ -520,6 +529,18 @@ func processLabelValues(valuesMap map[string]interface{}, clusterLabels map[stri
 			} else {
 				return fmt.Errorf("invalid_label_reference %s in key %s", valStr, key)
 			}
+		} else {
+			switch val.(type) {
```

Comment:
We already gather `valStr` on line 523. Perhaps, we can refactor this code to use it instead. I'd prefer that we handle this scenario once.

Reply:
I actually have an updated PR I was going to send in that changes how a lot of this stuff is done and does not have any extra string conversions in that area of the code.

After playing around with the configmap stuff in 0.3.6 I thought it would be fun to allow the value substitution to come from any overridden values like the ones imported from a configmap, not just the ones in helm.vlaues from the fleet.yaml.  So I moved this code into the helm install function to do that.  I also added basic cluster annotation support on top of the cluster one to deal with values that can not be stored in an label. It's a bigger PR but adds a lot more flexibility that has been preventing us from using fleet for more deployments.